src-tauri/src/utils/path.rs

```diff
@@ -29,9 +29,26 @@ fn get_user_data_dir() -> Result<PathBuf, String> {
         Ok(data_dir)
     }
 
-    #[cfg(not(target_os = "macos"))]
+    #[cfg(target_os = "linux")]
+    {
+        let home = env::var("HOME").map_err(|_| "Failed to get HOME environment variable")?;
+        let data_dir = PathBuf::from(home)
+            .join(".local")
+            .join("share")
+            .join("OpenList Desktop");
+        fs::create_dir_all(&data_dir)
```

Comment:
IMO only the logic obtaining data dir varies between platforms that needs to be gated behind `target_os`.